core/tests/ecs/query.cpp

```diff
@@ -28,6 +28,14 @@ static Opt<std::tuple<Ts...>> queryOne(World& world, Entity entity)
     return query.at(entity);
 }
 
+template <typename... Ts>
+static std::size_t queryMatches(World& world)
+{
+    QueryData<Ts...> queryD{world, {}};
+    Query<Ts...> query{queryD.view()};
+    return query.count();
+}
+
```

Comment:
Instead of adding this new function, I would just replace the implementation of queryCount to use `query.count()` instead of iterating.